grpc/server/src/main/java/io/helidon/grpc/server/SslConfiguration.java

```diff
@@ -0,0 +1,193 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.grpc.server;
+
+import java.nio.file.Paths;
+
+import io.helidon.config.Config;
+
+public class SslConfiguration {
+
+    /**
+     * True if use jdk ssl implementation
+     */
+    private boolean jdkSSL;
+
+    /**
+     * The TLS certs file.
+     */
+    private String tlsCert;
+
+    /**
+     * The TLS key file.
+     */
+    private String tlsKey;
+
+    /**
+     * The TLS CA file.
+     */
+    private String tlsCaCert;
+
+    /**
+     * Create a new instance.
+     */
+    public SslConfiguration(boolean jdkSSL, String tlsCert, String tlsKey, String tlsCaCert){
+        this.jdkSSL = jdkSSL;
+        this.tlsCert = tlsCert;
+        this.tlsKey = tlsKey;
+        this.tlsCaCert = tlsCaCert;
+    }
+
+    /**
+     * Return true if use jdk ssl implementation
+     */
+    public boolean isJdkSSL(){
+        return jdkSSL;
+    }
+
+    /**
+     * Return TLS certs file.
+     *
+     * @return the TLS certs file
+     */
+    public String getTLSCerts() {
+        return tlsCert;
+    }
+
+    /**
+     * Return the TLS key file.
+     *
+     * @return the location of the TLS key file to use
+     */
+    public String getTLSKey() {
+        return tlsKey;
+    }
+
+    /**
+     * Return the TLS CA certs file.
+     *
+     * @return the TLS CA certs file
+     */
+    public String getTLSClientCerts() {
+        return tlsCaCert;
+    }
+
+    /**
+     * Return an instance of builder.
+     *
+     * @return  an instance of builder
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    /**
+     * Return an instance of builder based on the specified external config.
+     *
+     * @param config external config
+     * @return an instance of builder
+     */
+    public static Builder builder(Config config) {
+        return new Builder(config);
+    }
+
+    /**
+     * Create an instance of sslConfig from external configuration source.
+     *
+     * @param config external config
+     * @return an instance of sslconfig
+     */
+    public static SslConfiguration create(Config config) {
+        return builder(config).build();
+    }
+
+    /**
+     * Builds the configuration for ssl.
+     */
+    static class Builder implements io.helidon.common.Builder<SslConfiguration> {
+        private boolean jdkSSL;
+        private String tlsCert = null;
+        private String tlsKey = null;
+        private String tlsCaCert = null;
+
+        private Builder() {};
+
+        private Builder(Config config) {
+            if (config == null) {
+                return;
+            }
+
+        String path = config.get("path").asString().orElse(".");
+
+        String tlsCert = config.get("tlsCert").asString().orElse(null);
+        if (tlsCert != null) {
+            this.tlsCert = Paths.get(path, tlsCert).toAbsolutePath().toString();
```

Comment:
This can now become `this.tlsCert = path.resolve(tlsCert).toAbsolutePath().toString();` which is slightly better than `Paths.get` because it doesn't assume default file system (see Javadoc for details)

Reply:
This whole thing could also be rewritten in a more "functional" style, by leveraging methods on the `Optional`:
```
        String tlsCert = config.get("tlsCert").asString()
                                .map(path::resolve)
                                .map(Path::toAbsolutePath)
                                .map(Path::toString)
                                .orElse(null);
```
Not sure that's any better or cleaner, though -- it's just more "functional"...